### ISSUE TITLE
template sudo_user in playbook such that inventory variables can be used in it

### ISSUE THREAD

**Issue by leowmjw** [ansible/ansible#3483]
Environment: Debian 7 running using Vagrant inside VirtualBox on Windows 7.

Expectation: Ansible playbooks with a sudo_user variable defined inside group_vars/all will be correctly expanded and used in SSH connection to a AWS node.  The variables strangely does not get expanded.

New Observation: Apparently my colleagues using it inside a AWS node (rather than Vagrant node) works.  I have not had the chance to try it out yet.  My suspicion now is that there may be a hidden Windows newline when I pull the playbook rule from my git repo using Windows.
# Scenario

I have sudo_user defined in group_vars:
vagrant@ansible:/vagrant$ cat group_vars/all
sudo_user: admin
...
## 1) When I try to run the ansible-playbook with below (it FAILS):
# This playbook deploys the common components for all nodes in the site.
- name: COMMON - Deployment
  hosts: all
  user: ${sudo_user}
  sudo: yes
  
  roles:
  - common

<ec2-54-218-102-xxx.us-west-2.compute.amazonaws.com> ESTABLISH CONNECTION FOR USER: ${sudo_user}
## 2) When I try to run the ansible-playbook with below (it SUCCEEDS):
# This playbook deploys the common components for all nodes in the site.
- name: COMMON - Deployment
  hosts: all
  user: admin
  sudo: yes
  
  roles:
  - common

<ec2-54-218-102-xxx.us-west-2.compute.amazonaws.com> ESTABLISH CONNECTION FOR USER: admin

My playbooks work fine when using SSH password on a local Vagrant installation.
# Details:

vagrant@ansible:/vagrant$ ansible --version
ansible 1.3 (devel b2d881a899) last updated 2013/07/07 13:00:11 (GMT +000)
vagrant@ansible:/vagrant$ ansible-playbook  -i /tmp/hosts -vvv dev-common.yml

---

< PLAY [COMMON - Deployment] >

---

```
    \   ^__^
     \  (oo)\_______
        (__)\       )\/\
            ||----w |
            ||     ||
```

---

< GATHERING FACTS >

---

```
    \   ^__^
     \  (oo)\_______
        (__)\       )\/\
            ||----w |
            ||     ||
```

<ec2-54-218-102-xxx.us-west-2.compute.amazonaws.com> ESTABLISH CONNECTION FOR USER: ${sudo_user}
<ec2-54-218-102-xxx.us-west-2.compute.amazonaws.com> EXEC ['ssh', '-tt', '-q', '-o', 'ControlMaster=auto', '-o', 'ControlPersist=60s', '-o', 'ControlPath=/tmp/ansible-ssh-%h-%p-%r', '-o', 'Port=22', '-o', 'KbdInteractiveAuthentication=no',
'-o', 'PasswordAuthentication=no', '-o', 'User=${sudo_user}', '-o', 'ConnectTimeout=10', 'ec2-54-218-102-xxx.us-west-2.compute.amazonaws.com', "/bin/sh -c 'mkdir -p $HOME/.ansible/tmp/ansible-1373202144.04-33991115349494 && chmod a+rx $HOME/.ansible/tmp/ansible-1373202144.04-33991115349494 && echo $HOME/.ansible/tmp/ansible-1373202144.04-33991115349494'"]

Cheers,
Michael Leow

On Sunday, July 7, 2013 8:26:31 PM UTC+8, Michael DeHaan wrote:
Can you please test this on Ansible 1.3 (devel)?

Thanks!

On Sat, Jul 6, 2013 at 9:31 PM, Michael Leow michae...@icarasia.com wrote:
Hi Michael,

Was trying to parameterize the sudo_users into my playbooks (so I can use it both for Vagrant and AWS); but as you can see in scenario #1; it fails.  When I try to hard code the sudo user (in scenario #2); it works.

So the question is: what is the correct way to parameterize using the values I set in groups_vars/all so that it is correctly being used by the playbook.

BTW: Kudos on the Ansible design; compared to Puppet, it is so much easier to learn and build up :)

Cheers,
Michael Leow

On Sunday, July 7, 2013 3:39:48 AM UTC+8, Michael DeHaan wrote:
Can you summarize what the expected result and actual result are?

I'm failing to understand the problem of something succeeding, which is how I read your point #2 :)

On Sat, Jul 6, 2013 at 2:13 PM, Michael Leow michae...@icarasia.com wrote:
Hi,

Need help to figure out if the following is a bug with Ansible 1.2.2:
# Scenario

I have sudo_user defined in group_vars:
vagrant@ansible:/vagrant$ cat group_vars/all
sudo_user: admin
...
## 1) When I try to run the ansible-playbook with below (it FAILS):
# This playbook deploys the common components for all nodes in the site.
- name: COMMON - Deployment
  hosts: all
  user: ${sudo_user}
  sudo: yes
  
  roles:
  - common

<ec2-54-218-102-xxx.us-west-2.compute.amazonaws.com> ESTABLISH CONNECTION FOR USER: ${sudo_user}
## 2) When I try to run the ansible-playbook with below (it SUCCEEDS):
# This playbook deploys the common components for all nodes in the site.
- name: COMMON - Deployment
  hosts: all
  user: admin
  sudo: yes
  
  roles:
  - common

<ec2-54-218-102-xxx.us-west-2.compute.amazonaws.com> ESTABLISH CONNECTION FOR USER: admin

My playbooks work fine when using SSH password on a local Vagrant installation.

Is this a bug  or am I missing something?  Thanks!

Regards,
Michael Leow

**Comment by mpdehaan** [comment]
I somewhat remember the email thread, however this is very long and not well formatted from the copy/paste, can you please update this to just the actual bug?

**Comment by mpdehaan** [comment]
Never mind, I remember now, I believe this is about needing to template sudo_user in a playbook so inventory variables can be used in it

editing the ticket....

**Comment by bennojoy** [comment]
This is fixed in the latest head. please do check from your end
